daemon/container_unix.go

```diff
@@ -1007,13 +1013,15 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 		}
 	}()
 
-	if err := container.updateEndpointNetworkSettings(n, ep); err != nil {
+	if err = container.updateEndpointNetworkSettings(n, ep); err != nil {
```

Comment:
Yes. defers with `:=` is a bad combination and it has tripped us once in the past.